src/main/java/org/jabref/model/entry/identifier/DOI.java

```diff
@@ -234,6 +234,12 @@ public String getNormalized() {
         return doi;
     }
 
+    public boolean isCompareNotExact(DOI o2) {
+        String s1 = this.doi.replaceAll("[^\\w]", "");
```

Comment:
why is the "replace" here necessary? Maybe it would be worthwhile to add it to the other cleanup that's already happing in the constructor?

Reply:
Yes, maybe it is. I decided to do in this way in order to divide logics of creature, comparison and inaccurate comparison of DOI. This is associated with task context. The DOI standart is not strictly defined. Some special characters can serve as indicators or as special symbols.

Reply:
@KunAndrew Can you add a test for this?

We agree that the DOI standard is vague at some places. However, in the concrete case, only white spaces are replaced.

Whitespace in aribtrary locations lead to a non-found doi.

![image](https://user-images.githubusercontent.com/1366654/91971606-c5fed900-ed19-11ea-93ff-62b35d40d4ab.png)


Proposed solution: Can you put that cleanup into the constructor the next two weeks? 😇